libraries/AP_OpticalFlow/AP_OpticalFlow_MSP.h

```diff
@@ -0,0 +1,37 @@
+#pragma once
+
+#include "OpticalFlow.h"
+#include <AP_HAL/utility/OwnPtr.h>
+
+#if HAL_MSP_ENABLED
+
+class AP_OpticalFlow_MSP : public OpticalFlow_backend
+{
+public:
+    /// constructor
+    using OpticalFlow_backend::OpticalFlow_backend;
+
+    // initialise the sensor
+    void init() override {}
+
+    // read latest values from sensor and fill in x,y and totals.
+    void update(void) override;
+
+    // get update from msp
+    void handle_msp(const msp_opflow_sensor_t &pkt) override;
+
+    // detect if the sensor is available
+    static AP_OpticalFlow_MSP *detect(OpticalFlow &_frontend);
+
+private:
+
+    uint64_t prev_frame_us;             // system time of last message when update was last called
```

Comment:
I guess we really need the 64bit time?

Reply:
Microseconds are used in the other optical flow backends.

I can't see why these should be much different to the MAV backend, which uses &micro;s

Reply:
@rmackay9 can you elaborate on this, I copied this code from AP_OpticalFlow_MAV and assumed if was generally all right